src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -110,10 +110,14 @@ public function denormalize($data, string $type, string $format = null, array $c
         $defaultDateTimeFormat = $this->defaultContext[self::FORMAT_KEY] ?? null;
 
         if (null !== $defaultDateTimeFormat) {
-            $object = \DateTime::class === $type ? \DateTime::createFromFormat($defaultDateTimeFormat, $data, $timezone) : \DateTimeImmutable::createFromFormat($defaultDateTimeFormat, $data, $timezone);
-
-            if (false !== $object) {
-                return $object;
+            try {
+                $object = \DateTime::class === $type ? \DateTime::createFromFormat($defaultDateTimeFormat, $data, $timezone) : \DateTimeImmutable::createFromFormat($defaultDateTimeFormat, $data, $timezone);
+
+                if (false !== $object) {
+                    return $object;
+                }
+            } catch (\Throwable $t) {
+                throw NotNormalizableValueException::createForUnexpectedDataType('The data must be type of string.', $data, [Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
```

Comment:
why this message? can't this fail for another reason? doesn't $e->getMessage() contain useful feedback?

Reply:
You're right. changed it against $e->getMessage()